frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/create-sc/wizard-pages/donut-chart.tsx

```diff
@@ -32,6 +39,16 @@ export const DiscoveryDonutChart: React.FC<DiscoveryDonutChartProps> = ({ state,
   const nodes = getNodes(state.showNodesListOnLVS, state.nodeNamesForLVS, state.nodeNames);
   const { podStatusOuterRadius: radius } = calculateRadius(220);
 
+  React.useEffect(() => {
+    if (discoveriesLoaded && !discoveriesLoadError && discoveriesData.length) {
+      if (discoveriesData[0].status?.conditions[0]?.type === 'Available') {
+        dispatch({ type: 'setIsLoadingDonutChart', value: false });
+      } else {
+        dispatch({ type: 'setIsLoadingDonutChart', value: true });
+      }
+    }
+  }, [discoveriesData, discoveriesLoaded, discoveriesLoadError, dispatch]);
```

Comment:
Perhaps, we can avoid setting a state.
We can just keep local variables. Here are few examples for reference:
https://github.com/openshift/console/blob/538c5386804f1dacb22e367da641596b950d63a4/frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/storage-efficiency-card/storage-efficiency-card-item.tsx#L5

https://github.com/openshift/console/blob/538c5386804f1dacb22e367da641596b950d63a4/frontend/packages/noobaa-storage-plugin/src/components/data-consumption-card/performance-graph.tsx#L36

Reply:
@afreen23 I understood. I am looking into it.